var/spack/repos/builtin/packages/perl/package.py

```diff
@@ -93,6 +93,23 @@ def configure_args(self):
             '-Dloclibpth=' + self.spec['gdbm'].prefix.lib,
         ]
 
+        # Extensions are installed into their private tree via
+        # `INSTALL_BASE`/`--install_base` (see [1]) which results in a
+        # "predictable" installation tree that sadly does not match the
+        # Perl core's @INC structure.  This means that when activation
+        # merges the extension into the extendee[2], the directory tree
+        # containing the extensions is not on @INC and the extensions can
+        # not be found.
+        #
+        # This bit prepends @INC with the directory that is used when
+        # extensions are activated [3].
+        #
+        # [1] https://metacpan.org/pod/ExtUtils::MakeMaker#INSTALL_BASE
+        # [2] via the activate method in the PackageBase class
+        # [3] https://metacpan.org/pod/distribution/perl/INSTALL#APPLLIB_EXP
+        config_args.append('-Accflags=-DAPPLLIB_EXP=\\"' + join_path(
+                           self.prefix.lib, 'perl5') + '\\"')
```

Comment:
Hi @adamjstewart , the quoting is quite fiddly here, because the library path is inserted into C code as a quoted string by the preprocessor. I tried a few different expressions, and this is what eventually worked. If you can suggest something better though, I'd be happy to use it.